mypy/build.py

```diff
@@ -2004,9 +2004,14 @@ def parse_file(self) -> None:
                             self.path, os.strerror(ioerr.errno))],
                         module_with_blocker=self.id)
                 except (UnicodeDecodeError, DecodeError) as decodeerr:
-                    raise CompileError([
-                        "mypy: can't decode file '{}': {}".format(self.path, str(decodeerr))],
-                        module_with_blocker=self.id)
+                    if self.path.endswith('.pyd'):
```

Comment:
I think it would be less repetitive to have the string selected based on the file type and raise the error at the end:
```python
                    if self.path.endswith('.pyd'):
                        err = "mypy: stubgen does not support .pyd files: '{}'".format(self.path)
                    else:
                        err = "mypy: can't decode file '{}': {}".format(self.path, str(decodeerr))
                    raise CompileError([err], module_with_blocker=self.id)
```

Reply:
very reasonable